NitroxClient/Unity/Helper/CoroutineHelper.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections;
+
+namespace NitroxClient.Unity.Helper;
+
+public static class CoroutineHelper
+{
+    public static IEnumerator SafelyYieldEnumerator(IEnumerator enumerator, Action<Exception> exceptionCallback)
```

Comment:
We can definitely. This is a copy of its implementation but with a callback instead of the IOut reference. To me it seems a bit more straightforward to have the exception handling in a defined callback and in a if clause after the yield operation.